src/test/ui/for/for-loop-has-unit-body.rs

```diff
@@ -1,8 +1,7 @@
-// run-pass
 fn main() {
     // Check that the tail statement in the body unifies with something
     for _ in 0..3 {
-        unsafe { std::mem::uninitialized() }
+        unsafe { std::mem::uninitialized() } //~ ERROR type annotations needed
```

Comment:
`unsafe { std::mem::uninitialized() }` no longer needs to unify with `()`, so it cannot infer its type.